### PR TITLE
Compare positions, reset filters

### PR DESCRIPTION
**Pull request by mjoyce91**
Adds functionality from #117 :
- Adds functionality to compare positions by making use of the existing "Add/remove from comparison" button using local storage.
- When at least one position exists within the local storage comparison array, a new link and button appear in the Results page - one a link to Compare Positions which shows a side by side of the positions; and another a button to reset any selected comparisons to an empty array.
- New `/compare:ids` route that expects a list of positions numbers, such as `/compare/0005BA6,1001CM7`.
- Additionally, local storage `gets`/`sets` were externalized to `utilities.js` for use across multiple components and abstraction.

Adds functionality from #29 :
- Because of current site structure, this simply returns the user to the home page. By default, filters are not saved, so returning the home page implicitly meets the requirement for the search filters to be reset.
- The user must click the button twice, as the requirements state the user must receive a prompt to confirm that they want to clear their search filters. The button changes text from "Reset Filters" to "Are you sure", which upon clicking returns the user to the home page.

### CODE REVIEW

File: src/Components/CompareList/CompareList.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+class CompareList extends Component {
+
+  onChildToggle() {
+    this.forceUpdate();
+    this.props.onToggle();
+  }
+
+  render() {
+    const { compare } = this.props;
+    // TODO - move language list to its own component, as this is used in multiple places
```
Comment (burgwyn) [inline src/Components/CompareList/CompareList.jsx]: Can you make an issue for this?

File: src/Components/ResetFilters/ResetFilters/ResetFilters.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react';
```
Comment (burgwyn) [inline src/Components/ResetFilters/ResetFilters/ResetFilters.jsx]: This looks double-nested in the folder

Comment (burgwyn) [inline src/Components/ResetFilters/ResetFilters/ResetFilters.jsx]: src/Components/ResetFilters/ResetFilters/ResetFilters.jsx

File: src/Components/CompareList/CompareList.jsx (hunk 1 above)
Comment (mjoyce91) [inline src/Components/CompareList/CompareList.jsx]: Done

File: src/Components/ResetFilters/ResetFilters/ResetFilters.jsx (hunk 2 above)
Comment (mjoyce91) [inline src/Components/ResetFilters/ResetFilters/ResetFilters.jsx]: @burgwyn - I wanted to make use of a connected component which would handle the route change, and a child component to handle the presentation. Maybe they could be combined into the same file but still exist as separate classes? I named the parent/connected component `ResetFiltersConnect.jsx` but maybe there's a better naming convention.

**Pull request merged**